apps/src/gamelab/spritelab/worldCommands.js

```diff
@@ -20,9 +20,10 @@ export const commands = {
   },
 
   setBackgroundImage(img) {
-    let backgroundImage = this.loadImage(img);
-    backgroundImage.name = img;
-    spriteUtils.background = backgroundImage;
+    if (this._preloadedBackgrounds[img]) {
```

Comment:
Nit: this would throw if we ever got here before the promise resolves. I don't think this can happen in practice, but maybe worth defending against?

https://github.com/code-dot-org/code-dot-org/blob/9addfa160d22cd63a539d03945d00078c8a21e59/apps/src/gamelab/GameLabP5.js#L606-L608

```suggestion
    if (this._preloadedBackgrounds && this._preloadedBackgrounds[img]) {
```